diff --git a/packages/tabs/index.ts b/packages/tabs/index.ts
index 19adf23c..0a7c0e46 100644
--- a/packages/tabs/index.ts
+++ b/packages/tabs/index.ts
@@ -261,7 +261,7 @@ SmartComponent({
           // waiting transition end
           setTimeout(() => {
             this.setData({ skipTransition: false });
-          }, this.data.duration);
+          }, this.data.duration * 1000);
         }
       });
     },
diff --git a/packages/tabs/index.wxs b/packages/tabs/index.wxs
index 7456e9e0..f5d86f23 100644
--- a/packages/tabs/index.wxs
+++ b/packages/tabs/index.wxs
@@ -77,10 +77,8 @@ function cardStyle(data) {
     transform: 'translateX(' + data.lineOffsetLeft + 'px)',
     '-webkit-transform': 'translate(' + data.lineOffsetLeft + 'px, 0px)',
     'background-color': data.color,
-    'transition-duration': !data.skipTransition ? data.duration + 's' : null,
-    '-webkit-transition-duration': !data.skipTransition
-      ? data.duration + 's'
-      : null,
+    'transition-duration': data.skipTransition ? null : data.duration + 's',
+    '-webkit-transition-duration': data.skipTransition ? null : data.duration + 's'
   });
 }
 
